lib/swoosh/adapters/smtp/helpers.ex

```diff
@@ -102,21 +102,24 @@ if Code.ensure_loaded?(:mimemail) do
        content}
     end
 
-    defp prepare_attachment(%{filename: filename, path: path, content_type: content_type, type: attachment_type, headers: custom_headers}) do
+    defp prepare_attachment(%{filename: filename, path: path, content_type: content_type, type: attachment_type, headers: custom_headers, data: data}) do
       [type, format] = String.split(content_type, "/")
-      file = File.read!(path)
+      content = case data do
```

Comment:
Shall we have this logic in the `Attachment` module? Otherwise we'll end up duplicating it in all adapters.

Maybe we can populate `data` when the attachment is created?

A lot of our adapters are doing `path |> File.read! |> Base.encode64` to include the attachments in the request body.

Reply:
Do you mean adding this logic to new function in `Attachment`? Anyway you can create attachment using `%Swoosh.Attachment{content_type: "text/plain", data: "text text"}` but in this case how can you add the logic?

Reply:
Good point. Maybe just a function that takes an `Attachment` and returns the content for it then?

Reply:
Added a new function to `Attachment` module that retrieves content from struct